dev/com.ibm.ws.jaxrs.2.0.client_fat/test-applications/jaxrsclientstandalone/src/com/ibm/ws/jaxrs20/client/JAXRSClientStandalone/client/ClientTestServlet.java

```diff
@@ -132,4 +138,210 @@ public void testFlowProgram_ClientStandalone(Map<String, String> param, StringBu
         c.close();
         ret.append(res);
     }
+
+
+    static class ClientFunction implements Function<Client, Client> {
+        @Override
+        public Client apply(Client client) {
+            return client;
+        }
+    }
+
+    static class WebTargetFunction implements Function<Client, WebTarget> {
+        @Override
+        public WebTarget apply(Client client) {
+            return client.target("http://localhost:8010/dumpservice/test");
+        }
+    }
+
+    static class InvocationBuilderFunction implements Function<Client, Builder> {
+        @Override
+        public Builder apply(Client client) {
+            WebTarget t = client.target("http://localhost:8010/dumpservice/test");
+            return t.request();
+        }
+    }
+
+    static class InvocationFunction implements Function<Client, Invocation> {
+        @Override
+        public Invocation apply(Client client) {
+            WebTarget t = client.target("http://localhost:8010/dumpservice/test");
+            return t.request().buildGet();
+        }
+    }
+
+    static class AsyncInvokerFunction implements Function<Client, AsyncInvoker> {
+        @Override
+        public AsyncInvoker apply(Client client) {
+            WebTarget t = client.target("http://localhost:8010/dumpservice/test");
+            return t.request().async();
+        }
+    }
+
+    static class CompletionStageRxInvokerFunction implements Function<Client, Object> {
+        @Override
+        public Object apply(Client client) {
+            WebTarget t = client.target("http://localhost:8010/dumpservice/test");
+            Builder builder = t.request();
+            try {
+                Method m = builder.getClass().getMethod("rx");
+                return m.invoke(builder);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    /**
+     * This test makes sure that if an application doesn't call Client.close() on their
+     * JAXRS Client object that the runtime will not leak.  This was a problem in the past 
+     * and resolved by asking customers to update their application.  This was partially fixed
+     * by using a WeakHashMap in JAXRSClientImpl, but the clientsPerModule collection in JAXRSClientImpl
+     * still has a hard reference.  With the introduction of using weak references in clientsPerModule, 
+     * the liberty runtime now remove the weak reference object on the next time a Client is created
+     * or closed.  This test validates that that function works and will continue to work into the future.
+     */
+    public void testMemoryLeak_ClientStandalone(Map<String, String> param, StringBuilder ret) {
+        ClientBuilder cb = ClientBuilder.newBuilder();
+        Client c = cb.build();
+
+        Map<String, List<?>> clientsPerModule = null;
+        if (c.getClass().getName().equals("com.ibm.ws.jaxrs20.client.JAXRSClientImpl")) {
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [Eq: equals method compares class names rather than class objects (EQ_COMPARING_CLASS_NAMES)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#eq-equals-method-compares-class-names-rather-than-class-objects-eq-comparing-class-names).
This method checks to see if two objects are the same class by checking to see if the names of their classes are equal. You can have different classes with the same name if they are loaded by different class loaders. Just check to see if the class objects are the same.

Reply:
This is a test case, not product code.  These tests are reused for Java EE7, EE8 and Jakarta EE9 testing.  The JAXRS code is different in Jakarta EE9 which is why this code is here.  Only if the class is that name will it actually use the test code that is done here.  I cannot use the Class object itself to see if it is there or not, because when testing with Jakarta EE9, I will end up with a NoClasDefFoundError because that class it not available when using the Jakarta EE9 JAXRS feature.